doc/cephfs/kernel.rst

```diff
@@ -1,58 +1,95 @@
+#. Copy the Ceph keyring from the MON host to client host at ``/etc/ceph``::
 
-Omitting the monitor addresses will cue ``mount.ceph`` to look them up
-in the local configuration file.  If you know at least one of the monitor
-addresses and ports you can specify them directly in a comma-separated
-list, and the kernel will avoid looking them up::
+    # on client host
+    mkdir /etc/ceph
+    scp {user}@{server-machine}:/etc/ceph/ceph.keyring /etc/ceph/ceph.keyring
 
-        # mount -t ceph 192.168.0.1:6789,192.168.0.2:6789:/ /mnt/mycephfs
+#. Ensure that the keyring has appropriate permissions::
 
-To mount a subtree of the cephfs root, append the path to the device
-string ::
+    chmod 600 /etc/ceph/ceph.keyring
 
-        # mount -t ceph :/subvolume/dir1/dir2  /mnt/mycephfs -o name=fs
+# It is recommeneded to have mount.ceph helper installed. Verify that it is
+  installed::
 
-To mount the Ceph file system with ``cephx`` authentication enabled, the
-kernel must authenticate with the cluster. The default ``name=`` option
-is ``guest``.  The mount.ceph helper will automatically attempt to find
-a secret key in a cephx keyring if it's not specified. For example, to
-mount the filesystem as the cephx user ``fs``::
+    ls /sbin/mount.ceph
 
-        # mount -t ceph :/ /mnt/mycephfs -o name=fs
+# You may also get the Ceph configuration file on the client host, although
+  this is optional::
 
-The secret can also be specified manually with the ``secret=`` option.::
+    ssh {user}@{mon-host} "sudo ceph config generate-minimal-config" | sudo tee /etc/ceph/ceph.conf
 
-	# mount -t ceph :/ /mnt/mycephfs -o name=fs,secret=AQATSKdNGBnwLhAAnNDKnH65FmVKpXZJVasUeQ==
+In general, your command to mount CephFS via kernel looks like this::
 
-For legacy usage, mount.ceph can be told to read a lone secret from a
-file. For example::
+    mount -t ceph {device-string}{path-to-mounted} {mount-point} -o {key-value-args} {other-args}
 
-	# mount -t ceph :/ /mnt/mycephfs -o name=fs,secretfile=/etc/ceph/admin.secret
+Mounting CephFS
+---------------
+On Ceph clusters, CephX is enabled by default. Use ``mount`` command to
+mount CephFS with the Kernel driver::
 
-See `User Management`_ for details on cephx.
+    mkdir /mnt/mycephfs
+    mount -t ceph :/ /mnt/mycephfs -o name=admin
 
-If you have more than one file system, specify which one to mount using
-the ``mds_namespace`` option, e.g.::
+The key-value argument right after option ``-o`` is CephX credential;
+``name`` is the username of the CephX user we are using to mount CephFS.
+The default value for ``name`` is ``guest``. You may also create a CephX user
+instead of using ``admin``.
 
-        # mount -t ceph :/ -o name=fs,mds_namespace=myfs
+In truth, the kernel driver also requires MON's socket and the secret key for
+the Cephx user. In case of above command ``mount.ceph`` helper figures out
+these details automatically. You can pass these details yourself too, although
+not mandatory::
```

Comment:
"it's not mandatory if you have a ceph.conf and keyring files"

Reply:
Rephrased this part.